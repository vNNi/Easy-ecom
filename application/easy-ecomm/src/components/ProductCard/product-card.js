import { useState } from 'react';

import { HeartOutlined, HeartFilled } from '@ant-design/icons';

import { currencyFormatter } from '../../utils';
import {
  Container,
  IconContainer,
  ImageContainer,
  InfosContainer,
  Title,
  InStockLabel,
  StockValue,
  Price,
} from './product-card.styles';

const ProductCard = ({
  id,
  title,
  price,
  image = 'https://ninajohansson.se/wp-content/themes/koji/assets/images/default-fallback-image.png',
  stock_count: stockCount,
  onClick = () => ({}),
  onFavorite = () => ({}),
  isFavorited,
}) => {
  const [favorited, setFavorited] = useState(isFavorited);
  const handleFavoriteClick = () => {
    onFavorite({ id, title, price, image, stock_count: stockCount });
    setFavorited(true);
  };

  const icon = favorited ? (
    <HeartFilled style={{ color: 'red' }} onClick={handleFavoriteClick} />
  ) : (
    <HeartOutlined onClick={handleFavoriteClick} />
  );
  return (
    <Container onClick={e => onClick(e, id)}>
      <IconContainer>{icon}</IconContainer>
      <ImageContainer>
        <img alt="" src={image} width={120} height={100} />
      </ImageContainer>
      <InfosContainer>
        <Title>{title}</Title>
        <div>
          <InStockLabel>Em estoque: </InStockLabel>
          <StockValue>{stockCount || 0}</StockValue>
        </div>
        <Price>{currencyFormatter({ value: Number(price) })}</Price>
      </InfosContainer>
    </Container>
  );
};

export default ProductCard;
