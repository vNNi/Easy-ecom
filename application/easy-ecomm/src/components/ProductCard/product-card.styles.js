import styled from 'styled-components';

export const Container = styled.h1`
  color: ${({ theme }) => theme.colors.primary};
`;
