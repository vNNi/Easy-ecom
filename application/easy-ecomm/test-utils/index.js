import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import '@testing-library/jest-dom/extend-expect'; // jest-dom custom assertions
import 'jest-styled-components';

import Theme from '../src/styles/theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
