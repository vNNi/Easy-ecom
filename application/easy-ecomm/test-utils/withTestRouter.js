import React from 'react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';

export default function withTestRouter(tree, router = {}) {
  const {
    route = '',
    pathname = '',
    pagename = '',
    pagetype = '',
    query = {},
    asPath = '',
    push = async () => true,
    replace = async () => true,
    reload = () => null,
    back = () => null,
    prefetch = async () => undefined,
    beforePopState = () => null,
    isFallback = false,
    events = {
      emit: () => null,
      off: () => null,
      on: () => null,
    },
  } = router;

  return (
    <RouterContext.Provider
      value={{
        asPath,
        back,
        beforePopState,
        events,
        isFallback,
        pagename,
        pagetype,
        pathname,
        prefetch,
        push,
        query,
        reload,
        replace,
        route,
      }}
    >
      {tree}
    </RouterContext.Provider>
  );
}
