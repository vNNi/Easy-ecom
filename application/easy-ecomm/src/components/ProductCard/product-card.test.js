import { renderWithTheme } from '../../../test-utils';

import ProductCard from './product-card';

describe('[Components] - product card', () => {
  test('should render correctly', () => {
    const comp = renderWithTheme(<ProductCard />);
    expect(comp).toMatchSnapshot();
  });
});
