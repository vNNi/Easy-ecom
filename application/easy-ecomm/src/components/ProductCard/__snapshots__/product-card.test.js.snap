// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Components] - product card should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 250px;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  padding: 0.5em;
}

.c0:hover {
  box-shadow: 0px 2px 12px rgba(81,81,81,0.12);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 span {
  font-size: 20px;
}

.c1:hover span {
  font-size: 22px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  margin: 0.5em 0;
  color: #404040;
  font-size: 16px;
}

.c3 {
  padding: 0 1em;
}

.c5 {
  font-size: 14px;
  color: #404040;
}

.c6 {
  color: #ffffff;
  background-color: #001529;
  padding: 2px 10px;
  border-radius: 20%;
  text-align: center;
  font-weight: bold;
}

.c7 {
  font-size: 20px;
  font-weight: bold;
  margin: 0.5em 0;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <span
            aria-label="heart"
            class="anticon anticon-heart"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="heart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
              />
            </svg>
          </span>
        </div>
        <div
          class="c2"
        >
          <img
            alt=""
            height="100"
            src="https://ninajohansson.se/wp-content/themes/koji/assets/images/default-fallback-image.png"
            width="120"
          />
        </div>
        <div
          class="c3"
        >
          <h2
            class="c4"
          />
          <div>
            <span
              class="c5"
            >
              Em estoque: 
            </span>
            <span
              class="c6"
            >
              0
            </span>
          </div>
          <p
            class="c7"
          >
            R$ NaN
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product-cardstyles__Container-sc-19h0a0q-0 inoYmN"
    >
      <div
        class="product-cardstyles__IconContainer-sc-19h0a0q-1 evYGRr"
      >
        <span
          aria-label="heart"
          class="anticon anticon-heart"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="heart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
            />
          </svg>
        </span>
      </div>
      <div
        class="product-cardstyles__ImageContainer-sc-19h0a0q-2 iOefdA"
      >
        <img
          alt=""
          height="100"
          src="https://ninajohansson.se/wp-content/themes/koji/assets/images/default-fallback-image.png"
          width="120"
        />
      </div>
      <div
        class="product-cardstyles__InfosContainer-sc-19h0a0q-4 jthKlt"
      >
        <h2
          class="product-cardstyles__Title-sc-19h0a0q-3 bqoOJr"
        />
        <div>
          <span
            class="product-cardstyles__InStockLabel-sc-19h0a0q-5 fubCiK"
          >
            Em estoque: 
          </span>
          <span
            class="product-cardstyles__StockValue-sc-19h0a0q-6 iHYHWm"
          >
            0
          </span>
        </div>
        <p
          class="product-cardstyles__Price-sc-19h0a0q-7 bObfeN"
        >
          R$ NaN
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
